jpsonic-main/src/test/java/com/tesshu/jpsonic/service/scanner/MediaScannerServiceImplTest.java

```diff
@@ -516,6 +525,163 @@ void testUpdateAlbum() {
+            assertEquals(1, genres.get(8).getAlbumCount());
+            assertEquals("GENRE_H", genres.get(9).getName());
+            assertEquals(1, genres.get(9).getAlbumCount());
+
+            // GENRE_I, GENRE_J has been deleted.
+
+            assertEquals("GENRE_K", genres.get(10).getName());
+            assertEquals(2, genres.get(10).getAlbumCount());
+            assertEquals("GENRE_L", genres.get(11).getName());
+            assertEquals(2, genres.get(11).getAlbumCount());
+
+            // ### Test UD(Tag Update&Delete)
+            Files.delete(Path.of(tempDir.toString(), "ARTIST1/ALBUM1/FILE01.mp3"));
+            FileUtils.copyFile(new File(resolveBaseMediaPath("Scan/MultiGenreCRUD/ARTIST1/ALBUM1/FILE01.mp3")),
+                    Path.of(tempDir.toString(), "ARTIST1/ALBUM1/FILE01.mp3").toFile());
+            Files.delete(Path.of(tempDir.toString(), "ARTIST1/ALBUM7/FILE11.mp3"));
+            FileUtils.copyFile(new File(resolveBaseMediaPath("Scan/MultiGenreCRUD/ARTIST1/ALBUM7/FILE11.mp3")),
+                    Path.of(tempDir.toString(), "ARTIST1/ALBUM7/FILE11.mp3").toFile());
+            TestCaseUtils.execScan(mediaScannerService);
+
+            // Deleting a file will reduce the number of genres by 2.
+            genres = searchService.getGenres(criteria, 0, Integer.MAX_VALUE);
+            assertEquals(13, genres.size());
+
+            assertEquals("Audiobook - Historical", genres.get(0).getName());
+            assertEquals(1, genres.get(0).getAlbumCount());
+            assertEquals("Audiobook - Sports", genres.get(1).getName());
+            assertEquals(1, genres.get(1).getAlbumCount());
+
+            assertEquals("GENRE_A-CHANGED", genres.get(2).getName());
+            assertEquals(1, genres.get(2).getAlbumCount());
+
+            // GENRE_A -> GENRE_A-CHANGED
+
+            assertEquals("GENRE_B", genres.get(3).getName());
+            assertEquals(1, genres.get(3).getAlbumCount());
+            assertEquals("GENRE_C", genres.get(4).getName());
+            assertEquals(1, genres.get(4).getAlbumCount());
+            assertEquals("GENRE_D", genres.get(5).getName());
+            assertEquals(1, genres.get(5).getAlbumCount());
+            assertEquals("GENRE_E", genres.get(6).getName());
+            assertEquals(1, genres.get(6).getAlbumCount());
+            assertEquals("GENRE_F", genres.get(7).getName());
+            assertEquals(1, genres.get(7).getAlbumCount());
+            assertEquals("GENRE_G", genres.get(8).getName());
+            assertEquals(1, genres.get(8).getAlbumCount());
+            assertEquals("GENRE_H", genres.get(9).getName());
+            assertEquals(1, genres.get(9).getAlbumCount());
+            assertEquals("GENRE_K", genres.get(10).getName());
+            assertEquals(2, genres.get(10).getAlbumCount());
+
+            assertEquals("GENRE_L", genres.get(11).getName());
+            assertEquals(2, genres.get(10).getAlbumCount()); // (2 -x-> 1)
+
```

Comment:
Test case seems to be incorrect! In the correct case it will probably work as expected.

Reply:
Done.